xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioDSP.h

```diff
@@ -0,0 +1,1298 @@
+    /// @return The needed size of output array or 0 if no changes within it
+    /// @remarks Optional. Is set by AE_DSP_ADDON_CAPABILITIES and asked with
+    /// GetCapabilities
+    ///
+    virtual unsigned int MasterProcessNeededSamplesize(const ADDON_HANDLE handle) { return 0; }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// @brief Returns the time in seconds that it will take
+    /// for the next added packet to be returned to KODI.
+    /// @param handle identification data for stream
+    /// @return the delay in seconds
+    /// @remarks Optional. Is set by AE_DSP_ADDON_CAPABILITIES and asked with
+    /// GetCapabilities
+    ///
+    virtual float MasterProcessGetDelay(const ADDON_HANDLE handle) { return 0.0f; }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// @brief Returns the from selected master mode performed channel alignment
+    /// @param handle identification data for stream
+    /// @retval out_channel_present_flags the exact channel present flags after
+    /// performed up-/downmix
+    /// @return the amount channels
+    /// @remarks Optional. Must be used and set if a channel up- or downmix is
+    /// processed from the active master mode
+    ///
+    virtual int MasterProcessGetOutChannels(const ADDON_HANDLE handle, unsigned long& out_channel_present_flags) { return 0; }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// @brief Master processing becomes performed with it
+    /// Here a channel up-mix/down-mix for stereo surround sound can be performed
+    /// Only one DSP add-on is allowed to-do this!
+    /// @param handle identification data for stream
+    /// @param array_in Pointer to input data memory
+    /// @param array_out Pointer to output data memory
+    /// @param samples Amount of samples inside array_in
+    /// @return Amount of samples processed
+    /// @remarks Optional. Is set by AE_DSP_ADDON_CAPABILITIES and asked with
+    /// GetCapabilities
+    ///
+    virtual unsigned int MasterProcess(const ADDON_HANDLE handle, const float** array_in, float** array_out, unsigned int samples) { return 0; }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// Used to get a information string about the processed work to show on skin
+    /// @return A string to show
+    /// @remarks Optional. Is set by AE_DSP_ADDON_CAPABILITIES and asked with
+    /// GetCapabilities
+    ///
+    virtual std::string MasterProcessGetStreamInfoString(const ADDON_HANDLE handle) { return ""; }
```

Comment:
@AlwinEsch
Will the std::string return value passed through our C interface boundaries?